blackboxopt/base.py

```diff
@@ -133,14 +137,20 @@ def __init__(
         super().__init__(search_space=search_space, seed=seed)
         self.objective = objective
 
-    def report_evaluation(self, evaluation: Evaluation) -> None:
-        raise_on_unknown_or_incomplete(
-            exception=ObjectivesError,
-            known=[self.objective.name],
-            reported=evaluation.objectives.keys(),
-        )
+    def report_evaluations(
+        self, evaluations: Union[Evaluation, Iterable[Evaluation]]
+    ) -> None:
+        if isinstance(evaluations, Evaluation):
+            evaluations = [evaluations]
+
+        for evaluation in evaluations:
+            raise_on_unknown_or_incomplete(
+                exception=ObjectivesError,
+                known=[self.objective.name],
+                reported=evaluation.objectives.keys(),
+            )
 
-        super().report_evaluation(evaluation)
+        super().report_evaluations(evaluations)
```

Comment:
By the way: Am I correct that this super call doesn't make any sense? As the `super()` class is in this case an not implemented abstract class anyway?

Reply:
Or what this done on purpose, @LGro ?